conda_smithy/configure_feedstock.py

```diff
@@ -595,9 +598,97 @@ def _load_forge_config(forge_dir, variant_config_files):
+    available_versions = sorted(available_versions, key=VersionOrder)
+    most_recent_version = available_versions[-1]
+    if installed_version is None:
+        msg = "{} is not installed in root env.".format(name)
+    elif VersionOrder(installed_version) < VersionOrder(most_recent_version):
+        msg = "{} version in root env is out-of-date.".format(name)
+    else:
+        return
+    if error_on_warn:
+        raise RuntimeError("{} Exiting.".format(msg))
+    else:
+        print(msg)
+
+
+def commit_changes(forge_file_directory, commit, cs_ver, cfp_ver):
+    if cfp_ver:
+        msg = 'Re-rendered with conda-smithy {} and pinning {}'.format(cs_ver, cfp_ver)
+    else:
+        msg = 'Re-rendered with conda-smithy {}'.format(cs_ver)
+    print(msg)
+
+    is_git_repo = os.path.exists(os.path.join(forge_file_directory, ".git"))
+    if is_git_repo:
+        has_staged_changes = subprocess.call(
+            [
+                "git", "diff", "--cached", "--quiet", "--exit-code"
+            ],
+            cwd=forge_file_directory
+        )
+        if has_staged_changes:
+            if commit:
+                git_args = [
+                    'git',
+                    'commit',
+                    '-m',
+                    'MNT: {}'.format(msg)
+                ]
+                if commit == "edit":
+                    git_args += [
+                        '--edit',
+                        '--status',
+                        '--verbose'
+                    ]
+                subprocess.check_call(
+                    git_args,
+                    cwd=forge_file_directory
+                )
+                print("")
+            else:
+                print(
+                    'You can commit the changes with:\n\n'
+                    '    git commit -m "MNT: {}"\n'.format(msg)
+                )
+            print("These changes need to be pushed to github!\n")
+        else:
+            print("No changes made. This feedstock is up-to-date.\n")
+
+
+def main(forge_file_directory, no_check_uptodate, commit):
+    error_on_warn = False if no_check_uptodate else True
+    index = conda_build.conda_interface.get_index(channel_urls=['conda-forge']) 
+    r = conda_build.conda_interface.Resolve(index)
+
+    # Check that conda-smithy is up-to-date
+    check_version_uptodate(r, "conda-smithy", __version__, error_on_warn)
+
     forge_dir = os.path.abspath(forge_file_directory)
-    config = _load_forge_config(forge_dir, variant_config_files)
+    print(os.listdir(os.path.join(forge_dir, "recipe")))
+    print(os.path.join(forge_dir, "recipe"))
+
+    exclusive_config_file_default = os.path.join(forge_dir, "recipe", "conda_build_config.yaml")
+    config = _load_forge_config(forge_dir, exclusive_config_file_default)
+    exclusive_config_file = config['exclusive_config_file']
+
+    # Don't check for conda-forge-pinning if there is one in the recipe
```

Comment:
Just cut this section.  You want to default to conda-forge-pinning, but allow people to specify a different config file with the same mentality as conda-forge-pinning.  That is, another config file that represents the broad, general pinnings to apply to almost everything.

Think of config files in recipes as fine-tuning or overrides on that broad file.  For example, it might be used to force mingw compilers instead of the standard VS stuff on windows.  It's a per-recipe override mechanism.

Reply:
> ...allow people to specify a different config file with the same mentality as conda-forge-pinning.

Having an option to override `conda-forge-pinning` is a good idea. Currently the user can accomplish this by specifying `exclusive_config_file` in `conda-forge.yml`. Does that satisfy what you had in mind or do you know of use cases where having a CLI option would help?

Reply:
Seems fine to me to change it in conda-forge.yml.  My advice would be to draw a very clear line that if you do so, your config should NOT be alongside meta.yaml.  Reserve the contents of the recipe folder for "overrides," and designate a different location for exclusive_config_files.

The net effect might be roughly equivalent, but because exclusive_config_file goes at the front of any order, and the other goes at the end (implicitly), I'd advise some conceptual purity here, just in case there are subtle issues that arise.

Maybe I'm misreading this code, but I'm understanding it as "use conda_build_config.yaml as exclusive_config_file if it is in the recipe folder."  That's what I'm discouraging - not the addition of exclusive_config_file as a parameter in conda-forge.yml

Reply:
Sure. I think we are agreeing on how `exclusive_config_file` should be used.

Though based on your recent comment am now thinking we might want to drop `exclusive_config_file` from `conda-forge.yml`. If we are wanting to override `exclusive_config_file`, maybe it is best to stick with a CLI option to override (as you had suggested in your previous comment). Having an option in `conda-forge.yml` might be too easily abused.

Reply:
Raised this as issue ( https://github.com/conda-forge/conda-smithy/issues/674 ). That way we can come to some decision before the 3.0.0 release, but not belabor this PR with it. Should be easy to update the code either way we go.